v2/.gitignore

```diff
@@ -4,3 +4,7 @@ config/crd/out
 config/crd/kustomization.yaml
 config/webhook/manifests.yaml
 config/rbac/role.yaml
+
+# Helm chart manifest
+charts/azure-service-operator/templates/generated
```

Comment:
minor: You can nest `.gitignore` so you could put this closer to the actual files being ignored (like in `charts/azure-service-operator`?). Is the reason to avoid that that then that file would get included in the actual helm chart? If so this makes sense and is fine, but if that's not an issue then might wanna put this closer to what it's ignoring?